### PR TITLE
feat: adds useNProgress function

### PR DESCRIPTION
**Pull request by anteriovieira**
I would like to know from the community if it makes sense to have this function here.

> Reactive wrapper for [`nprogress`](https://github.com/rstacruz/nprogress)

```js {6}
import { useNProgress } from '@vueuse/integrations'

const { isLoading } = useNProgress()

function toggle() {
  isLoading.value = !isLoading.value
}
```

useProgress

### CODE REVIEW

File: packages/integrations/useProgressBar/index.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+const loading = ref(false)
+
+useProgressBar(loading)
+
+function toggle() {
+  loading.value = !loading.value
+}
+```
+
```
Comment (antfu) [inline packages/integrations/useProgressBar/index.md:33]: Maybe we should have some statements showing that the existence the `options` arg and redirect users to check `nprogress`'s documentation.

File: packages/integrations/useProgressBar/index.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+/**
+ * Reactive progress bar.
+ *
+ * @see   {@link https://vueuse.js.org/useProgressBar}
+ * @param newIcon
+ * @param options
+ */
+export function useProgressBar(
+  current: MaybeRef<boolean | null | undefined> = null,
```
Comment (antfu) [inline packages/integrations/useProgressBar/index.ts:13]: Thinking we should use numbers to set the progress.

https://github.com/rstacruz/nprogress#advanced-usage

File: packages/integrations/useProgressBar/index.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    : ref<boolean | null>(current)
+
+  if (options)
+    progress.configure(options)
+
+  watch(loading, active => active ? progress.start() : progress.done())
+
+  tryOnUnmounted(progress.remove)
+
+  return loading
```
Comment (antfu) [inline packages/integrations/useProgressBar/index.ts:27]: Maybe it's better to return an object giving more controls to users.

```ts
{
  progress: Ref<number>,
  isLoading: Ref<boolean>,
  remove: Fn,
  done: Fn,
  /* ... */
}
```

**Comment by matias-capeletto** [comment]
Nice! Maybe it could be called `useNProgress()` so it is clear that this is a wrapper for NProgress?

**Comment by antfu** [comment]
Agree with `useNProgress` tho it reads weird but make more sense 👍

File: packages/integrations/useProgressBar/index.md (hunk 1 above)
Comment (anteriovieira) [reply packages/integrations/useProgressBar/index.md]:
```js {13}
export function useNProgress(
  currentProgress: MaybeRef<number | null | undefined> = null,
  options?: NProgressOptions | undefined,
) {
  const progress = isRef(currentProgress) ? currentProgress : ref(null)
  const isLoading = ref(false)

  if (options)
    nprogress.configure(options)

  watch(isLoading, active => active ? nprogress.start() : nprogress.done())

  watch(progress, (n) => {
    if (typeof n === 'number')
      nprogress.set(n)
  })

  tryOnUnmounted(nprogress.remove)

  return {
    isLoading,
    progress,
    start: nprogress.start,
    done: nprogress.done,
    remove: nprogress.remove,
  }
}
```

What do you think?

File: packages/integrations/useProgressBar/index.ts (hunk 2 above)
Comment (anteriovieira) [reply packages/integrations/useProgressBar/index.ts]:
```js {13}
export function useNProgress(
  currentProgress: MaybeRef<number | null | undefined> = null,
  options?: NProgressOptions | undefined,
) {
  const progress = isRef(currentProgress) ? currentProgress : ref(null)
  const isLoading = ref(false)

  if (options)
    nprogress.configure(options)

  watch(isLoading, active => active ? nprogress.start() : nprogress.done())

  watch(progress, (n) => {
    if (typeof n === 'number')
      nprogress.set(n)
  })

  tryOnUnmounted(nprogress.remove)

  return {
    isLoading,
    progress,
    start: nprogress.start,
    done: nprogress.done,
    remove: nprogress.remove,
  }
}
```

What do you think?

File: packages/integrations/useProgressBar/index.ts (hunk 3 above)
Comment (anteriovieira) [reply packages/integrations/useProgressBar/index.ts]: I think we have a problem, when I use the library functions to change the status I don't have any events that can notify me to update the refs

File: packages/integrations/useNProgress/index.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+### Passing a source ref
+
+You can pass a `ref` to it, changes from of the source ref will be reflected to your loading progress.
+
+```js {4}
+import { ref } from 'vue'
+import { useNProgress } from '@vueuse/integrations'
+
+const isLoading = ref(false)
```
Comment (antfu) [inline packages/integrations/useNProgress/index.md:25]: Do we need to update the docs? I think we are accepting numbers now, or do you think we should support boolean as well?

Comment (anteriovieira) [reply packages/integrations/useNProgress/index.md]: no, I don't think we need to support boolean here

**Pull request merged**